### PR TITLE
Subgraph updates to development

### PR DESCRIPTION
**Pull request by codenamejason**
Jason and Daniele working branch to update development

### CODE REVIEW

**Comment by codenamejason** [comment]
This PR also has all the wagmi and rainbow kit updates as well as the subgraph updates

File: client/src/components/application/Form.tsx [hunk 1]
```diff
@@ -73,8 +74,18 @@ export default function Form({
   const handleSubmitApplication = async () => {
     setSubmitted(true);
     await validate();
-    if (formValidation.valid) {
-      dispatch(submitApplication(round.address, formInputs));
+    if (roundManagerClient && formValidation.valid) {
+      showToast(true);
+      await submitApplication(roundManagerClient, round.address, formInputs)
```
Comment (tim-schultz) [inline client/src/components/application/Form.tsx]: Maybe stick to either `await` or `.then()` here?

File: client/src/components/base/Preview.tsx [hunk 2]
```diff
@@ -1,75 +1,68 @@
...
 
   const publishProject = async () => {
+    if (!chain || !signer) {
+      console.error("Missing chain or signer");
+      return;
+    }
     setSubmitted(true);
     showToast(true);
-    await dispatch(publishGrant(currentProjectId));
+    await publishGrant(formInputs, chain!.id, signer, currentProjectId)
```
Comment (tim-schultz) [inline client/src/components/base/Preview.tsx]: Same here with `async` vs `then`  🙏

File: client/src/components/base/Preview.tsx [hunk 3]
```diff
@@ -1,75 +1,68 @@
...
-import { formatDate } from "../../utils/components";
 import { slugs } from "../../routes";
+import { FormInputs, ProjectFormStatus } from "../../types";
+import { formatDate } from "../../utils/components";
 import Details from "../grants/Details";
 import Button, { ButtonVariants } from "./Button";
 import Toast from "./Toast";
 import TXLoading from "./TXLoading";
-import { ProjectFormStatus } from "../../types";
-import { formReset } from "../../actions/projectForm";
```
Comment (tim-schultz) [inline client/src/components/base/Preview.tsx:14]: Should we delete `actions/projectForm` and `reducers/projectForm` files if it is no longer being used?

File: client/src/components/base/WalletDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        <MenuList>
+          {/* <MenuItem minH="48px" onClick={() => setOpen(true)}>
+            <span>Switch Network</span>
+          </MenuItem> */}
+          <MenuItem minH="40px" onClick={() => disconnect()}>
+            <span>Disconnect</span>
+          </MenuItem>
+        </MenuList>
+      </Menu>
+      {/* todo: redo this modal using our base setup
```
Comment (tim-schultz) [inline client/src/components/base/WalletDisplay.tsx]: Should we wipe these comments out if?

File: client/src/components/rounds/Show.tsx [hunk 5]
```diff
@@ -1,95 +1,104 @@
...
-      dispatch(loadRound(roundId));
+  async function fetchRound() {
+    if (!roundManagerClient) return;
+    const roundInfo = await useFetchRoundByAddress(
+      roundManagerClient,
+      roundId!
+    );
+
+    if (!roundInfo) {
+      console.error("Cannot load round", roundId);
```
Comment (tim-schultz) [inline client/src/components/rounds/Show.tsx:36]: @michellema1208 @willsputra Maybe we can get designs for unsuccessful load of round data? Also for loading projects?

File: client/src/components/rounds/Show.tsx [hunk 6]
```diff
@@ -1,95 +1,104 @@
...
+      return;
     }
-  }, [dispatch, roundId]);
-
-  if (props.web3ChainId !== props.roundChainId) {
```
Comment (tim-schultz) [inline client/src/components/rounds/Show.tsx:45]: I added this a couple days ago to indicate that the round contract is on a different network than the one the user is connected to https://github.com/gitcoinco/grant-hub/pull/243 Any chance we could add this back in?

File: client/src/components/base/WalletDisplay.tsx (hunk 4 above)
Comment (codenamejason) [reply client/src/components/base/WalletDisplay.tsx]: cleaned up

**Comment by gravityblast** [comment]
@codenamejason I noticed we no longer "auto-login" users if they already gave permissions to the dapp. 
We added it in the past as a feature and it's also much better in development to avoid having to login at each refresh :)
can we add it back please?

**Comment by gravityblast** [comment]
when I create a new project and click on "save and publish", I no longer see the messages in the toast component that were showing "uploading metadata", "uploading images", "waiting to confirm tx", etc, but I see an empty toast:

Screenshot from 2022-08-16 10-26-37

**Comment by gravityblast** [comment]
I can't apply to the round `0x3879a696253Aa2FaE4164C21e16104894dE632c9` anymore, but we don't show any error, just the loading spinner and a message in the console `Cannot load round 0x3879a696253Aa2FaE4164C21e16104894dE632c9`

Screenshot from 2022-08-16 10-33-06

**Comment by gravityblast** [comment]
> @codenamejason I noticed we no longer "auto-login" users if they already gave permissions to the dapp.
We added it in the past as a feature and it's also much better in development to avoid having to login at each refresh :)
can we add it back please?

I notice I'm "logged-out" often even while I make some changes on the code so I would definitely put it back to have development quicker like before ;)

**Comment by gravityblast** [comment]
if I go to a round page and then back to the dashboard I see this dialog:

Screenshot from 2022-08-16 10-52-02

But I think it should be there only when I try to apply to a round and I'm redirected to create a project because I didn't have one yet.
In my case I already had many so I don't think we should show it.

**Comment by gravityblast** [comment]
I also noticed it remains there every time I go to the dashboard and if I click to the apply button it redirects to the route without "compiling" the path `#/chains/:chainId/rounds/:roundId/apply`

**Comment by gravityblast** [comment]
when I click on kovan or goerli to switch network but then I don't accept, I see an empty page with `Something went wrong.`, maybe we don't catch the error. I would catch the error without doing anything so that the user can stay in the current network.

File: client/src/actions/rounds.ts [hunk 7]
```diff
@@ -75,195 +67,118 @@ const loadingError = (address: string, error: string): RoundsActions => ({
...
-      ...projectQuestion,
-      id: projectQuestionId,
+export const loadRound =
+  (roundInfo: RoundResponse) => async (dispatch: Dispatch) => {
+    if (
+      !roundInfo ||
+      roundInfo.round.roundMetaPtr.pointer === undefined ||
+      roundInfo.round.applicationMetaPtr.pointer === undefined
+    ) {
+      // TODO: log error and metric
```
Comment (gravityblast) [inline client/src/actions/rounds.ts]: shall we dispatch an error to be able to show it in the UI?

**Comment by gravityblast** [comment]
I tried to apply to the latest deployed round at `/#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3`.
With this one I can see the first page, but when I click on "apply to this round" the form doesn't load and I see this error in the console ` TypeError: roundApplication.applicationSchema is undefined`

**Comment by gravityblast** [comment]
I see we some calls in the console to both alchemy and infura, are they both needed?

**Comment by gravityblast** [comment]
I noticed we have a call to the subgraph to get all the projects, but we also call the subgraph for each project id.
I see the first call returns already everything, shall we remove all the next calls?

**Comment by gravityblast** [comment]
do we actually need the network switch? it's useful for development but in theory in production we should just stay on the network where the project registry is deployed /cc @michellema1208

**Review by gravityblast — CHANGES_REQUESTED** [review]
great job @codenamejason  @DanieleSalatti !
I added some comments on some things we need to fix before merging.

now that all the logic is in the react components and we don't rely anymore on the current state we should 
find the best way to add tests without having to mock all the services we have. 
we should maybe focus on this in the next sprint I think.

great job!!!
:rocket:

File: client/src/components/rounds/Show.tsx (hunk 5 above)
Comment (willsputra) [reply client/src/components/rounds/Show.tsx:36]: @schultztimothy @michellema1208 added the error states here!
https://www.figma.com/file/obBzCCL8KwoTlVBb5g3ccN/Grant-Hub---Wireframes?node-id=2189%3A13013

File: client/src/components/application/Form.tsx (hunk 1 above)
Comment (DanieleSalatti) [reply client/src/components/application/Form.tsx]: Picking this up

File: client/src/components/base/Preview.tsx (hunk 2 above)
Comment (DanieleSalatti) [reply client/src/components/base/Preview.tsx]: Picking this up

File: client/src/components/base/Preview.tsx (hunk 3 above)
Comment (DanieleSalatti) [reply client/src/components/base/Preview.tsx:14]: Picking this up

**Comment by DanieleSalatti** [comment]
> I noticed we have a call to the subgraph to get all the projects, but we also call the subgraph for each project id.
> I see the first call returns already everything, shall we remove all the next calls?

Picking this up.

Edit: https://github.com/codenamejason/grant-hub/pull/6

**Comment by DanieleSalatti** [comment]
> I also noticed it remains there every time I go to the dashboard and if I click to the apply button it redirects to the route without "compiling" the path `#/chains/:chainId/rounds/:roundId/apply`

This should be fixed in the next update to this PR

**Comment by DanieleSalatti** [comment]
I am adding a 404:

Schermata 2022-08-16 alle 10 51 38

Since we changed the URL for the rounds and I keep going to the old links, which don't handle the error properly.

File: client/src/actions/rounds.ts (hunk 7 above)
Comment (codenamejason) [reply client/src/actions/rounds.ts]: > do we actually need the network switch? it's useful for development but in theory in production we should just stay on the network where the project registry is deployed /cc @michellema1208

I removed it from the header where the chain is displayed.

Comment (codenamejason) [reply client/src/actions/rounds.ts]: Also set up env to only allow OP mainnet for prod for OP launch and our test chains for dev env

Comment (codenamejason) [reply client/src/actions/rounds.ts]: this was meant for the previous comment...

**Comment by codenamejason** [comment]
> > @codenamejason I noticed we no longer "auto-login" users if they already gave permissions to the dapp.
> > We added it in the past as a feature and it's also much better in development to avoid having to login at each refresh :)
> > can we add it back please?
> 
> I notice I'm "logged-out" often even while I make some changes on the code so I would definitely put it back to have development quicker like before ;)

updated to auto login if you have connected previously

File: client/src/components/rounds/Show.tsx (hunk 5 above)
Comment (DanieleSalatti) [reply client/src/components/rounds/Show.tsx:36]: I will address this in a separate PR if that's ok.

**Comment by DanieleSalatti** [comment]
> I tried to apply to the latest deployed round at `` /#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3``. With this one I can see the first page, but when I click on "apply to this round" the form doesn't load and I see this error in the console `` TypeError: roundApplication.applicationSchema is undefined`

I cannot find that in the graph...

Here: https://thegraph.com/hosted-service/subgraph/thelostone-mc/program-factory-v0

The query:

```
{
  round(id: "0xb91fec0b68f39cbfdd75e4f08042c60724e1bd3") {
      id
      applicationsStartTime
      applicationsEndTime
      applicationMetaPtr {
        id
        pointer
        protocol
      }
      roundMetaPtr {
        id
        pointer
        protocol
      }
  }
}
```

Gives:

```
{
  "data": {
    "round": null
  }
}
```

File: client/src/actions/rounds.ts (hunk 7 above)
Comment (DanieleSalatti) [reply client/src/actions/rounds.ts]: > shall we dispatch an error to be able to show it in the UI?

This is coming as well

File: client/src/components/rounds/Show.tsx (hunk 6 above)
Comment (DanieleSalatti) [reply client/src/components/rounds/Show.tsx:45]: Yep, on it!

**Comment by codenamejason** [comment]
I think we got everything @gravityblast and @schultztimothy commented on.

**Comment by gravityblast** [comment]
after going to `/#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3b`, and then to the dashboard, I still see the "Time to get your project funded" dialog, which should only showed if I went to the round application without having a project, and I'm redirected to create a new project.

**Comment by gravityblast** [comment]
if I go to a non existing round like `/#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3c`, I no longer see an error message but just loading gif and an error in the console

Screenshot from 2022-08-16 10-33-06

**Comment by gravityblast** [comment]
I'm still not able to see the "loading" states we had before when I create a project:

Screenshot from 2022-08-16 10-26-37

**Comment by gravityblast** [comment]
> I cannot find that in the graph...

@DanieleSalatti yeah I guess it was an old one and not indexed, we should put back the error messages to at least say something like `round not found`

File: client/src/components/rounds/Show.tsx (hunk 5 above)
Comment (codenamejason) [reply client/src/components/rounds/Show.tsx:36]: added error modal to open on error of round data

**Comment by tim-schultz** [comment]
> I'm still not able to see the "loading" states we had before when I create a project:
> 
> Screenshot from 2022-08-16 10-26-37

Looks like this one was introduced outside of this PR with the Preview step. I created this to resolve it: https://github.com/gitcoinco/grant-hub/pull/258

**Comment by DanieleSalatti** [comment]
Closing in favour of #262, which contains all the changes from this PR in a conveniently squashed commit (+ 1 commit for the merge itself and conflict resolution).

**Pull request closed without merging**